ee/clickhouse/migrations/0026_persons_and_groups_on_events.py

```diff
@@ -0,0 +1,40 @@
+from infi.clickhouse_orm import migrations
+
+from posthog.client import sync_execute
+from posthog.settings import CLICKHOUSE_CLUSTER
+from posthog.settings.data_stores import CLICKHOUSE_DATABASE, CLICKHOUSE_REPLICATION
+
+ADD_COLUMNS_BASE_SQL = """
+ALTER TABLE {table}
+ON CLUSTER '{cluster}'
+ADD COLUMN IF NOT EXISTS person_id UUID,
+ADD COLUMN IF NOT EXISTS person_properties VARCHAR,
+ADD COLUMN IF NOT EXISTS group0_properties VARCHAR,
+ADD COLUMN IF NOT EXISTS group1_properties VARCHAR,
+ADD COLUMN IF NOT EXISTS group2_properties VARCHAR,
+ADD COLUMN IF NOT EXISTS group3_properties VARCHAR,
+ADD COLUMN IF NOT EXISTS group4_properties VARCHAR;
+"""
+
+operations = [
+    migrations.RunSQL(ADD_COLUMNS_BASE_SQL.format(table="events", cluster=CLICKHOUSE_CLUSTER)),
+]
+
+
+if CLICKHOUSE_REPLICATION:
+
+    # CLICKHOUSE_REPLICATION may be turned on but the instance might still
+    # be on the old schema if it didn't run the sharding async migration (0004_replicated_schema)
+    verify_table_exists_result = sync_execute(
+        f"""
+            SELECT count(*) from system.tables WHERE database='{CLICKHOUSE_DATABASE}' and name='writable_events'
```

Comment:
We already have a similar check in code. Extract `from ee.clickhouse.materialized_columns.replication import clickhouse_is_replicated` to a new module and use that.